core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -234,6 +252,7 @@ private class ShuffleStatus(
     for (mapIndex <- mapStatuses.indices) {
       if (mapStatuses(mapIndex) != null && f(mapStatuses(mapIndex).location)) {
         _numAvailableMapOutputs -= 1
+        mapStatusesDeleted(mapIndex) = mapStatuses(mapIndex)
```

Comment:
This has potential for interacting badly with correctness changes, right ?
See `DAGScheduler.submitMissingTasks` when stage is Indeterminate

Reply:
In that case, `mapId` is not the same, isn't it? We are reusing with `mapId` at [line 170](https://github.com/apache/spark/pull/32730/files#diff-a3b15298f97577c1fadcc2d76d015eebd6343e246c6717417d33f3c458847f46R170), @mridulm .
```
val index = mapStatusesDeleted.indexWhere(x => x != null && x.mapId == mapId)
```

Reply:
BTW, I agree with you that we don't have a test coverage for the indeterministic stage case.

Reply:
You are right, the check on mapId and mapIndex ensure correctness. Thanks for clarifying !